tests/main/netplan-apply/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure that netplan apply works with network-setup-control
+
+details: |
+    Netplan apply is used to apply network configuration to the system
+
+# run on all classic ubuntu LTS and current dev systems 16.04+
+systems: 
+    - ubuntu-16*
+    - ubuntu-18*
+    - ubuntu-19*
+    - ubuntu-20*
+
+debug: |
+
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    snap install test-snapd-netplan-apply --edge
+
+    echo "temporarily installing netplan D-Bus service until SRU is done..."
+    # for now, until netplan's D-Bus has been SRU'd to the netplan.io debian 
+    # package, we will just manually install the netplan D-Bus service units
+    # into the machine from the snap and install netplan.io from the archive
+    apt update
+
+    # the netplan package was renamed nplan -> netplan.io in bionic
+    case "$SPREAD_SYSTEM" in
+        ubuntu-16*)
+            NETPLAN_APT_PKG=nplan;;
```

Comment:
s/nplan/netplan/ ?

Reply:
I booted xenial in multipass and:
```
multipass@venerated-newt:~$ apt-cache show nplan
N: Can't select versions from package 'nplan' as it is purely virtual
N: No packages found
multipass@venerated-newt:~$ apt-cache show netplan
N: Unable to locate package netplan
E: No packages found
multipass@venerated-newt:~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

Something is off here?

Reply:
`netplan.io` then?

Reply:
Ah, sorry, I didn't have the cache at all:
```

multipass@venerated-newt:~$ apt-cache show netplan
Package: netplan
Priority: optional
Section: universe/misc
Installed-Size: 155
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Petter Reinholdtsen <pere@debian.org>
Architecture: amd64
Source: plan
Version: 1.10.1-5
Replaces: plan (<< 1.7.1-1)
Depends: adduser (>= 3.21), libc6 (>= 2.15)
Suggests: plan
Conflicts: plan (<< 1.7.1-1)
Filename: pool/universe/p/plan/netplan_1.10.1-5_amd64.deb
Size: 29448
MD5sum: c7330e5b7d6a3faf47252a045ee2ad3b
SHA1: 23a20803bb8cc67aa8e025c58e71b4456ed4b157
SHA256: c5dbdc212adea7545709e2f7c1b0e4b6c34c6ef4956328876a8cc0077f204caa
Description-en: network server for `plan'
 Plan is a schedule planner based on X/Motif.
 .
 Netplan adds to plan multiuser capability using an IP server.
 .
 WARNING: the best level of authentication offered by netplan in
 this version is identd.  That's quite weak, so watch the manpage and
 tune the config carefully.
Description-md5: f65dc3e49b68b13ca3e8f766701f07c2
Homepage: http://www.bitrot.de/plan.html
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Origin: Ubuntu

multipass@venerated-newt:~$ apt-cache show nplan
Package: nplan
Architecture: amd64
Version: 0.32~16.04.6
Multi-Arch: foreign
Priority: standard
Section: universe/net
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 130
Depends: libc6 (>= 2.4), libglib2.0-0 (>= 2.16.0), libuuid1 (>= 2.16), libyaml-0-2, systemd (>= 229-4ubuntu20), python3, python3-yaml
Suggests: network-manager | wpasupplicant
Conflicts: netplan
Breaks: network-manager (<< 1.2.2-0ubuntu4~)
Filename: pool/universe/n/nplan/nplan_0.32~16.04.6_amd64.deb
Size: 41868
MD5sum: 4c43c1c1b6ad10d4e3293ff2a1a67b49
SHA1: 65c7a0344df8e0c566b3d5c5a407529913fafd77
SHA256: 46198816f72a65aaf6f24f853fadf06a978c97d3e961b0b2ea00acb136acd540
Description-en: YAML network configuration abstraction for various backends
 netplan reads YAML network configuration files which are written
 by administrators, installers, cloud image instantiations, or other OS
 deployments. During early boot it then generates backend specific
 configuration files in /run to hand off control of devices to a particular
 networking daemon.
 .
 Currently supported backends are networkd and NetworkManager.
Description-md5: 171990596b9adbbde53590d783767a6d
```

Reply:
Right as @zyga discovered, netplan in xenial is something else, we want `nplan` in xenial and `netplan.io` in bionic and later.